src/Instance.ts

```diff
@@ -250,11 +255,11 @@ export class Marked {
 
   #parseMarkdown(lexer: (src: string, options?: MarkedOptions) => TokensList | Token[], parser: (tokens: Token[], options?: MarkedOptions) => string) {
     return (src: string, options?: MarkedOptions | undefined | null): string | Promise<string> => {
-      const origOpt = { ...options };
-      const opt = { ...this.defaults, ...origOpt };
+      const origOpt: MarkedOptions = { ...options };
+      const opt: MarkedOptions = { ...this.defaults, ...origOpt };
 
       // Show warning if an extension set async to true but the parse was called with async: false
-      if (this.defaults.async === true && origOpt.async === false) {
+      if (isAsyncOptions(this.defaults) && isSyncOptions(origOpt)) {
         if (!opt.silent) {
           console.warn('marked(): The async option was set to true by an extension. The async: false option sent to parse will be ignored.');
```

Comment:
Question for maintainer: should this warning not be silent ?
Because as far as i know https://github.com/markedjs/marked/pull/3103 tried to "fix" the type to avoid unexepected Promises.

With this PR, the type returned by `marked()` and `marked.parse()` will have miss if a library change the default.async to true and user should be warned about it.

Reply:
The silent option is for when they want things to fail instead of get warnings. If they want to see the error they can just set the silent option to false.

Reply:
So you are suggesting throwing an error here instead of logging a massage? I think we could do that. This would need to be a breaking change.